subsys/net/lib/sockets/sockets.c

```diff
@@ -1161,15 +1177,17 @@ int z_impl_zsock_poll(struct zsock_pollfd *fds, int nfds, int timeout)
 				break;
 			}
 
-			if (timeout == K_NO_WAIT) {
+			if (K_TIMEOUT_EQ(timeout, K_NO_WAIT)) {
 				break;
 			}
 
-			if (timeout != K_FOREVER) {
-				/* Recalculate the timeout value. */
-				remaining_time = time_left(entry_time, timeout);
-				if (remaining_time <= 0) {
+			if (K_TIMEOUT_EQ(timeout, K_FOREVER)) {
```

Comment:
Looks like a missing `!` here since the original comparison was `timeout != K_FOREVER`

Reply:
Yep, fixed